pennylane/_qubit_device.py

```diff
@@ -592,6 +605,10 @@ def sample_basis_states(self, number_of_states, state_probability):
         shots = self.shots
 
         basis_states = np.arange(number_of_states)
+        if qml.math.ndim(state_probability) == 2:
+            return np.array(
+                [np.random.choice(basis_states, shots, p=prob) for prob in state_probability]
+            )
```

Comment:
`state_probability` always was one-dimensional, and two-dimensional `state_probability` is equivalent to broadcasting.
`np.random.choice` does not support broadcasting in the way we would require here, as far as I'm aware.